nautobot/core/templates/inc/nav_menu.html

```diff
@@ -13,795 +13,52 @@
-                        <li{% if not perms.extras.view_webhook %} class="disabled"{% endif %}>
-                            {% if perms.extras.add_webhook %}
-                                <div class="buttons pull-right">
-                                    <a href="{% url 'extras:webhook_add' %}" class="btn btn-xs btn-success" title="Add"><i class="mdi mdi-plus-thick"></i></a>
-                                </div>
-                            {% endif %}
-                            <a href="{% url 'extras:webhook_list' %}">Webhooks</a>
-                        </li>
-                    {% endif %}
-                {% endif %}
-
-                {% if perms.extras.view_customlink or not settings.HIDE_RESTRICTED_UI %}
-                        <li class="divider"></li>
-                        <li class="dropdown-header">Miscellaneous</li>
-
-                    {% if perms.extras.view_customlink or not settings.HIDE_RESTRICTED_UI %}
-                        <li{% if not perms.extras.view_customlink %} class="disabled"{% endif %}>
-                            {% if perms.extras.add_customlink %}
-                                <div class="buttons pull-right">
-                                    <a href="{% url 'extras:customlink_add' %}" class="btn btn-xs btn-success" title="Add"><i class="mdi mdi-plus-thick"></i></a>
-                                </div>
-                            {% endif %}
-                            <a href="{% url 'extras:customlink_list' %}">Custom Links</a>
-                        </li>
-                    {% endif %}
-                {% endif %}
-                    </ul>
-                </li>
+                <ul class="nav navbar-nav">
+                    {% for tab_name, tab_details in registry.nav_menu.tabs.items %}
+                        {% if request.user|has_perms:tab_details.permissions or not settings.HIDE_RESTRICTED_UI %}
+                            <li class="dropdown">
+                                <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">{{ tab_name }} <span class="caret"></span></a>
+                                <ul class="dropdown-menu">
+                                    {% for group_name, group_details in tab_details.groups.items %}
+                                        {% if request.user|has_perms:group_details.permissions or not settings.HIDE_RESTRICTED_UI %}
+                                            <li class="dropdown-header">{{ group_name }}</li>
+                                            {% for item_link, item_details in group_details.items.items %}
+                                                {% if not item_details.permissions or request.user|has_perms:item_details.permissions %}
+                                                    {% if request.user|has_perms:item_details.permissions or not settings.HIDE_RESTRICTED_UI %}
```

Comment:
Not clear to me why these two nested ifs instead of a single combined one?

```suggestion
                                                {% if not item_details.permissions or request.user|has_perms:item_details.permissions or not settings.HIDE_RESTRICTED_UI %}
```